src/widgets/input_float.rs

```diff
@@ -0,0 +1,158 @@
+use crate::{
+    types::{Vector2, Rect, Color},
+	hash,
+    widgets::Editbox,
+    Id, Layout, Ui,
+};
+
+#[derive(Default)]
+struct State {
+	string_represents: f32,
+	string: String,
+	before: String,
+	drag: Option<Drag>,
+}
+
+#[derive(Clone, Copy)]
+struct Drag {
+	start_mouse: Vector2,
+	start_data: f32,
+}
+
+pub struct InputFloat<'a> {
+    id: Id,
+    label: &'a str,
+    size: Option<Vector2>,
+	step: f32,
+}
+
+impl<'a> InputFloat<'a> {
+    pub fn new(id: Id) -> InputFloat<'a> {
+        InputFloat {
+            id,
+            size: None,
+            label: "",
+			step: 0.1,
+        }
+    }
+
+    pub fn label<'b>(self, label: &'b str) -> InputFloat<'b> {
+        InputFloat {
+            label,
+            id: self.id,
+            size: self.size,
+            step: self.step,
+        }
+    }
+
+    pub fn size(self, size: Vector2) -> Self {
+        Self {
+            size: Some(size),
+            ..self
+        }
+    }
+
+	/// Ratio of pixels on the x-axis dragged to how much the value should be changed.
+	/// Default is `0.1`.
+    pub fn step(self, step: f32) -> Self {
+        Self {
+            step,
+            ..self
+        }
+    }
+
+    pub fn ui(self, ui: &mut Ui, data: &mut f32) {
+        let context = ui.get_active_window_context();
+		let state_hash = hash!(self.id, "input_float_state");
+        let mut s: State = std::mem::take(
+			context
+				.storage_any
+				.get_or_default(state_hash)
+		);
+
+        let size = self.size.unwrap_or_else(|| {
+            Vector2::new(
+                context.window.cursor.area.w
+                    - context.global_style.margin * 2.
+                    - context.window.cursor.ident,
+                19.,
+            )
+        });
+        let pos = context.window.cursor.fit(size, Layout::Vertical);
+
+        let editbox_area = Vector2::new(
+			if self.label.is_empty() { size.x } else { size.x / 2.0 },
+			size.y,
+		);
+
+        let editbox = Editbox::new(self.id, editbox_area)
+            .position(pos)
+            .multiline(false);
+
+        let hovered = Rect::new(pos.x, pos.y, editbox_area.x, editbox_area.y)
+            .contains(context.input.mouse_position);
+
+        if hovered && context.input.is_mouse_down() {
+            s.drag = Some(Drag {
+                start_mouse: context.input.mouse_position,
+                start_data: *data,
+            });
+            context.window.input_focus = Some(self.id);
+            context.input.cursor_grabbed = true;
+        }
+
+        if let Some(drag) = s.drag {
+            if context.input.is_mouse_down == false {
+                s.drag = None;
+                context.input.cursor_grabbed = false;
+                if !hovered {
+                    context.window.input_focus = None;
+                } else {
+                    use super::editbox::EditboxState;
```

Comment:
I am not quite sure that exposing editboxes private state is a really good idea. 
Maybe we could add some configuration to Editbox'es public api instead?